### ISSUE TITLE
Tiler out of bounds on large geotiff

### ISSUE THREAD

**Issue by HeDo88TH** [DroneDB/DroneDB#257]
Some large geotiffs throw an out of bounds error. 
Test datasets: https://testhub.dronedb.app/r/floridadronetech/serenoalakes and https://testhub.dronedb.app/r/floridadronetech/orangecounty

Here it is the output of the `ddb tile` command:

```
C:\Users\IT 2\Pictures\test-tiff>ddb tile OrangeCounty20210624.tif test
2021-07-24 01:58:56.008 VERB  [15832] [main@56] DDB v0.9.10 (git commit cb82649)
2021-07-24 01:58:56.008 VERB  [15832] [main@57] SQLite version: 3.24.0
2021-07-24 01:58:56.009 VERB  [15832] [main@58] SpatiaLite version: 4.3.0-RC1
2021-07-24 01:58:56.010 VERB  [15832] [main@59] GDAL version: 3.0.4
2021-07-24 01:58:56.010 VERB  [15832] [main@60] CURL version: libcurl/7.70.0-DEV OpenSSL/1.1.1g zlib/1.2.3
2021-07-24 01:58:56.033 DEBUG [15832] [ddb::GDALTiler::GDALTiler@140] Bounds (output SRS): -9.08973e+06,3.2978e+06,-9.08893e+06,3.29836e+06
2021-07-24 01:58:56.033 DEBUG [15832] [ddb::GDALTiler::GDALTiler@150] MinZ: 15
2021-07-24 01:58:56.034 DEBUG [15832] [ddb::GDALTiler::GDALTiler@151] MaxZ: 22
2021-07-24 01:58:56.034 DEBUG [15832] [ddb::GDALTiler::GDALTiler@152] Num bands: 3
2021-07-24 01:58:56.034 DEBUG [15832] [ddb::Tiler::getMinMaxCoordsForZ@76] MinMaxCoordsForZ(15) = (8951, 19080), (8952, 19080)
2021-07-24 01:58:56.035 DEBUG [15832] [ddb::Tiler::getTilesForZoomLevel@62] 8951 19080 15
2021-07-24 01:58:56.035 DEBUG [15832] [ddb::Tiler::getTilesForZoomLevel@62] 8952 19080 15
2021-07-24 01:58:56.036 DEBUG [15832] [ddb::TilerHelper::runTiler@188] Tiling 8951 19080 15
2021-07-24 01:58:56.036 DEBUG [15832] [ddb::Tiler::getMinMaxCoordsForZ@76] MinMaxCoordsForZ(15) = (8951, 19080), (8952, 19080)
2021-07-24 01:58:56.037 DEBUG [15832] [ddb::GDALTiler::tile@186] GeoQuery: 0,0|21948x27298|161,10|95x118
2021-07-24 02:00:02.630 DEBUG [15832] [ddb::GDALTiler::tile@262] Wrote tile data
2021-07-24 02:00:02.630 DEBUG [15832] [ddb::GDALTiler::tile@274] Wrote tile alpha
test\15\8951\19080.png
2021-07-24 02:00:02.646 DEBUG [15832] [ddb::TilerHelper::runTiler@188] Tiling 8952 19080 15
2021-07-24 02:00:02.647 DEBUG [15832] [ddb::Tiler::getMinMaxCoordsForZ@76] MinMaxCoordsForZ(15) = (8951, 19080), (8952, 19080)
2021-07-24 02:00:02.647 DEBUG [15832] [ddb::GDALTiler::tile@186] GeoQuery: 21947,0|16886x27298|0,10|73x118
2021-07-24 02:01:15.103 DEBUG [15832] [ddb::GDALTiler::tile@262] Wrote tile data
2021-07-24 02:01:15.103 DEBUG [15832] [ddb::GDALTiler::tile@274] Wrote tile alpha
test\15\8952\19080.png
2021-07-24 02:01:15.114 DEBUG [15832] [ddb::Tiler::getMinMaxCoordsForZ@76] MinMaxCoordsForZ(16) = (17903, 38160), (17904, 38161)
2021-07-24 02:01:15.115 DEBUG [15832] [ddb::Tiler::getTilesForZoomLevel@62] 17903 38160 16
2021-07-24 02:01:15.115 DEBUG [15832] [ddb::Tiler::getTilesForZoomLevel@62] 17904 38160 16
2021-07-24 02:01:15.116 DEBUG [15832] [ddb::Tiler::getTilesForZoomLevel@62] 17903 38161 16
2021-07-24 02:01:15.116 DEBUG [15832] [ddb::Tiler::getTilesForZoomLevel@62] 17904 38161 16
2021-07-24 02:01:15.117 DEBUG [15832] [ddb::TilerHelper::runTiler@188] Tiling 17903 38160 16
2021-07-24 02:01:15.118 DEBUG [15832] [ddb::Tiler::getMinMaxCoordsForZ@76] MinMaxCoordsForZ(16) = (17903, 38160), (17904, 38161)
2021-07-24 02:01:15.119 DEBUG [15832] [ddb::GDALTiler::tile@186] GeoQuery: 0,27243|21949x55|66,0|190x0
Geoquery out of bounds
```

Could it be that the files are corrupted or that the tiler has run out of memory? @pierotofy

**Comment by HeDo88TH** [comment]
Here it is the exception line: https://github.com/DroneDB/DroneDB/blob/master/src/gdaltiler.cpp#L283
Failing condition: https://github.com/DroneDB/DroneDB/blob/master/src/gdaltiler.cpp#L190
`g.w.ysize` is `0`

**Comment by pierotofy** [comment]
Size is not the key issue here, the lat/lon X/Y coordinates were swapped when computing the orthophoto extent.

This is addressed in #258 

Size will affect performance however, it's currently very slow to tile a GeoTIFF without overviews. 

Speed will be addressed with #259
